src/library_html5.js

```diff
@@ -112,6 +112,10 @@ var LibraryHTML5 = {
     },
     
     canPerformEventHandlerRequests: function() {
+      if (navigator.userActivation) {
+        // Use transient activation status for browsers that support it
+        return navigator.userActivation.isActive;
+      }
```

Comment:
How about

```js
  canPerformEventHandlerRequests: function() {
  // Verify against transient activation status from UserActivation API
  // whether it is possible to perform a request here without needing to defer. See
  // https://developer.mozilla.org/en-US/docs/Web/Security/User_activation#transient_activation
  // and https://caniuse.com/mdn-api_useractivation
  return navigator.userActivation?.isActive
    || (JSEvents.inEventHandler && JSEvents.currentEventHandler.allowsDeferredCalls);
}
```
It would be interesting to experiment with a micro code-size improvement to remove `JSEvents.inEventHandler` and `JSEvents.currentEventHandler.allowsDeferredCalls` altogether when targeting only browsers that are guaranteed to support UserActivation API.

Reply:
We don't generally allow optional chaining in the generated JS yet, but perhaps we should.  If we can show that closure that lower it to ES5 as needed that would be great... let me see if I can confirm that now.

Reply:
I've added the comment lines. Let me know when you've decided about the optional chaining operator.

Reply:
I think we can land this now and consider it as part of the motivation for landing #19772

Reply:
Sorry, I'm not sure I understand. Do you mean using the optional chaining in the PR and wait for #19772 completion to merge this PR? Or the opposite (not using the optional chaining here)?

Reply:
I meant that I think we should land this PR now, as is